edb/schema/scalars.py

```diff
@@ -222,7 +222,45 @@ class ScalarTypeCommand(
     schema_metaclass=ScalarType,
     context_class=ScalarTypeCommandContext,
 ):
-    pass
+    def apply(
```

Comment:
It's better to overload `validate_create()` and `validate_alter()` instead of `apply()` (you almost never want to overload `apply()`).

Reply:
This isn't why I did it that way originally, but that approach doesn't catch the case where an existing abstract base has a new concrete base added (since the AlterScalarType that updates ancestors is generated after `validate_alter` is called, I think).

I think that we could have validate_alter explicitly search all descendants when bases are added and explicitly do this check on them. Is that an improvement over overloading `apply()`?

Reply:
Yea, I think that would be a cleaner approach, I think.